### ISSUE TITLE
A way to cache reproducible extension results between bazel server restarts

### ISSUE THREAD

**Issue by jkurland-roku** [bazelbuild/bazel#24723]
### Which category does this issue belong to?

Core, External Dependency

### What underlying problem are you trying to solve with this feature?

Currently extensions marked as reproducible are rerun more than they need to since they aren't tracked in the lock file. Since some extensions are slow it would be useful to have a mechanism to skip running them.

### Which operating system are you running Bazel on?

ubuntu

### What is the output of `bazel info release`?

release 8.0.0

**Comment by fmeum** [comment]
https://docs.google.com/document/d/1TjA7-M5njkI1F38IC0pm305S9EOmxcUwaCIvaSmansg/edit?usp=drivesdk included an "always update" lockfile under the output base for reproducible extensions. We didn't implement it at the time simply because we didn't have examples of slow but reproducible extensions. Now that we do, I think that we should implement this.

@Wyverald @meteorcloudy What do you think?

**Comment by meteorcloudy** [comment]
I think it's a good idea, I guess `bazel clean --expunge` should still nuke it?

**Comment by fmeum** [comment]
Yes, it would behave similar to the local action cache and be stored directly under the output base.

**Comment by meteorcloudy** [comment]
Happy to review a PR!

**Comment by Wyverald** [comment]
SGTM!

> Now that we do, I think that we should implement this.

Just for the record, what are some examples of slow but reproducible extensions?

**Comment by fmeum** [comment]
`rules_rust`'s `crate_universe` is the example: https://bazelbuild.slack.com/archives/CSV56UT0F/p1734442667932509

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 8.3.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/8.3.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=8.3.0rc1. Thanks!
